backend/shelf-function/src/main/java/com/htec/shelffunction/controller/ExecuteController.java

```diff
@@ -0,0 +1,32 @@
+package com.htec.shelffunction.controller;
+
+import com.htec.shelffunction.service.ExecuteService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/execute")
+public class ExecuteController {
+
+    private final ExecuteService executeService;
+
+    public ExecuteController(ExecuteService executeService) {
+        this.executeService = executeService;
+    }
+
+    @GetMapping("/{lang}/{functionId}")
```

Comment:
maybe something like /language/{lang}/function/{functionId} so that caller can now which param is in which place